frontend/libs/health-department/feature-case-detail/src/lib/occasion/occasion-detail-dialog/occasion-detail-dialog.component.ts

```diff
@@ -0,0 +1,143 @@
+  selector: 'qro-occasion-detail-dialog',
+  templateUrl: './occasion-detail-dialog.component.html',
+  styleUrls: ['./occasion-detail-dialog.component.scss'],
+})
+export class OccasionDetailDialogComponent implements OnInit {
+  initialOccasion: OccasionDto;
+  occasionFormGroup: FormGroup;
+  timepickerTheme: NgxMaterialTimepickerTheme;
+
+  @Input()
+  occasion(occasion: OccasionDto) {
+    this.initialOccasion = occasion;
+  }
+
+  constructor(public dialogRef: MatDialogRef<OccasionDetailDialogComponent>, private builder: FormBuilder) {
+    this.initialOccasion = {
+      additionalInformation: '',
+      address: undefined,
+      contactPerson: '',
+      end: undefined,
+      occasionCode: '',
+      start: undefined,
+      title: '',
+      trackedCaseId: '',
+      visitorGroups: [],
+    };
+    this.occasionFormGroup = this.builder.group(this.mapOccasionToForm());
+    this.timepickerTheme = {
+      container: {
+        buttonColor: '#5ce1e6',
+      },
+      clockFace: {
+        clockHandColor: '#5ce1e6',
+      },
+      dial: {
+        dialBackgroundColor: '#5ce1e6',
+      },
+    };
+  }
+
+  ngOnInit() {
+    this.occasionFormGroup = this.mapOccasionToForm();
+    this.initializeValidators();
+  }
+
+  private mapTimestampToString(timestamp: Date) {
+    if (!timestamp) {
+      return '';
+    }
+    let timestampHour = timestamp?.getHours()?.toString();
+    timestampHour === '0' ? (timestampHour = '00') : (timestampHour = '0');
+    let timestampMinute = timestamp?.getMinutes()?.toString();
+    timestampMinute === '0' ? (timestampMinute = '00') : (timestampMinute = '0');
+    return timestampHour.concat(':').concat(timestampMinute);
+  }
+
+  setPickedStartTime(event: string) {
+    this.occasionFormGroup.controls.timeStart.setValue(event);
+  }
+
+  setPickedEndTime(event: string) {
+    this.occasionFormGroup.controls.timeEnd.setValue(event);
+  }
+
+  close() {
+    this.dialogRef.close();
+  }
+
+  closeAndSubmit() {
+    this.dialogRef.close(this.mapFormToOccasion());
+  }
+
+  private mapPickedDateAndTime(time: string, date: Date): Date {
+    const splitted = time.split(':');
+    const hour = splitted[0];
+    const minute = splitted[1];
+
+    if (!(date instanceof Date)) {
+      // @ts-ignore
+      date = (date as unknown).toDate();
+    }
+
+    hour ? date.setHours(Number(hour)) : date.setHours(0);
+    minute ? date.setMinutes(Number(minute)) : date.setMinutes(0);
+    return date;
+  }
+
+  private initializeValidators() {
+    this.occasionFormGroup.controls.title.setValidators([Validators.required]);
+    this.occasionFormGroup.controls.start.setValidators([Validators.required]);
+    this.occasionFormGroup.controls.end.setValidators([Validators.required]);
+    this.occasionFormGroup.controls.zipCode.setValidators([Validators.pattern('[0-9]{5}'), Validators.required]);
+    this.occasionFormGroup.controls.timeStart.setValidators([Validators.pattern('((?:(?:0|1)\\d|2[0-3])):([0-5]\\d)')]);
```

Comment:
vielleicht macht es Sinn, das Pattern zu den anderen Pattern zu packen und dort auch in die allgemeinen Fehlermeldungen aufzunehmen, so dass wir hier keine Sonderlocke haben